pd/src/verify.rs

```diff
@@ -64,11 +64,17 @@ impl StatelessTransactionExt for Transaction {
         for action in self.transaction_body().actions {
             match action {
                 Action::Output(output) => {
-                    if !output.body.proof.verify(
-                        output.body.value_commitment,
-                        output.body.note_commitment,
-                        output.body.ephemeral_key,
-                    ) {
+                    if output
+                        .body
+                        .proof
+                        .verify(
+                            output.body.value_commitment,
+                            output.body.note_commitment,
+                            output.body.ephemeral_key,
+                        )
+                        .is_err()
+                    {
+                        // TODO should the verification error be bubbled up here?
```

Comment:
As soon as we see an invalid proof in any action, we can bubble up the error as the entire transaction has failed